Embedded values documentation (#58)
moovweb-docs/xdn-docs: guides/routing.md

  .use(nextRoutes)
```

### Altering requests and responses

XDN offers APIs to manipulate request and response headers and cookies. The APIs are:

> | Operation     | Request                 | Upstream Response              | XDN Response           |

Reviewer: ```suggestion
| Operation     | Request                 | Upstream Response              | Response sent to Browser          |
```

|---------------|-------------------------|--------------------------------|------------------------|
| Set header    | `setRequestHeader`      | `setUpstreamResponseHeader`    | `setResponseHeader`    |
| Add cookie    | `*`                     | `addUpstreamResponseCookie`    | `addResponseCookie`    |
| Update header | `updateRequestHeader`   | `updateUpstreamResponseHeader` | `updateResponseHeader` |
| Update cookie | `*`                     | `updateUpstreamResponseCookie` | `updateResponseCookie` |
| Remove header | `removeRequestHeader`   | `removeUpstreamResponseHeader` | `removeResponseHeader` |
| Remove cookie | `*`                     | `removeUpstreamResponseCookie` | `removeResponseCookie` |

> `*` Adding, updating and removing a request cookie can all be achieved with `updateRequestHeader` applied to `cookie` header.

Reviewer: ```suggestion
`*` Adding, updating, or removing a request cookie can all be achieved with `updateRequestHeader` applied to `cookie` header.
```

Author: For this one we actually don't have specific APIs so it is either "all" or "" but not "also" (no alternative but those)

Reviewer: Ah, I misread.  Updated the suggestion.

> You can find detailed descriptions of these APIs at `@xdn/core` [documentation](/docs/api/core/classes/_router_responsewriter_.responsewriter.html)

Reviewer: ```suggestion
You can find detailed descriptions of these APIs in the `@xdn/core` [documentation](/docs/api/core/classes/_router_responsewriter_.responsewriter.html)
```

#### Embedded values

> The values that ar used for adding or updating are template literals that into which you can embed values from requests and responses. Such values are all embedded by using `${value}` format.

Reviewer: ```suggestion
You can inject values from the request or response into headers or cookies as template literals using the `${value}` format. For example: `setResponseHeader('original-request-path', '${path}')` would add an `original-request-path` response header whose value is the request path.
```

| Value           | Embedded value         | Description                                                          |
|-----------------|------------------------|----------------------------------------------------------------------|

> | HTTP method     | `${method}`            | The value of HTTP method used for the request (e.g. `GET`)           |

Reviewer: ```suggestion
| HTTP method     | `${method}`            | The value of the HTTP method used for the request (e.g. `GET`)           |
```
